### PR TITLE
SW-2934 Misc updates for nursery withdrawals in withdrawal log

### PR DESCRIPTION
**Pull request by karthikbtf**
- format table values
- parse formatted values in certain places, such as in reassignment for validation to work
- table sort by quantity is now fixed

Terraware App 2023-02-09 14-53-03

Terraware App 2023-02-09 14-29-43

Terraware App 2023-02-09 14-24-12

Terraware App 2023-02-09 14-23-42

### CODE REVIEW

File: src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx [hunk 1]
```diff
@@ -636,7 +636,7 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
                       id='withdrawnQuantity'
                       onChange={(value: unknown) => setWithdrawnQuantity(value as number)}
                       type='text'
-                      value={numericFormatter.format(withdrawnQuantity)}
+                      value={withdrawnQuantity}
```
Comment (karthikbtf) [inline src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx:639]: this was a bugfix... format was intended for the total ready quantity

File: src/components/NurseryWithdrawals/WithdrawalLogRenderer.tsx [hunk 2]
```diff
@@ -20,82 +20,93 @@ const useStyles = makeStyles((theme: Theme) => ({
...
       );
     }
-    return <CellRenderer index={index} column={column} row={row} value='' />;
-  }
 
-  return <CellRenderer {...props} />;
-}
+    if (column.key === 'totalWithdrawn') {
+      return <CellRenderer index={index} column={column} value={formatter.format(value as number)} row={row} />;
+    }
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/WithdrawalLogRenderer.tsx:80]: this was the only code added here..everything else that's shown as 'modified' was not modified.. it's a PR diff artifact

File: src/components/NurseryWithdrawals/WithdrawalLogRenderer.tsx [hunk 3]
```diff
@@ -20,82 +20,93 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
 }));
 
-export default function WithdrawalLogRenderer(props: RendererProps<TableRowType>): JSX.Element {
+const WithdrawalLogRenderer = (formatter: { format: (num: number) => string }) => {
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/WithdrawalLogRenderer.tsx:23]: made this a higher order function, causing most of the code below to be indented by 2 space.. showing up as modified..

File: src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx (hunk 1 above)
Comment (constanzauanini) [reply src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx:639]: So the rule is to format on "non editable" fields?

Comment (karthikbtf) [reply src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx:639]: Rule is to not format what the user is typing. Other display values should be formatted I think that’s the plan. You had also called it out in an earlier PR. We need to do this in all the accessions views as well.

File: src/components/NurseryWithdrawals/NurseryWithdrawals.tsx [hunk 4]
```diff
@@ -199,10 +207,15 @@ export default function NurseryWithdrawals(): JSX.Element {
     );
     if (apiSearchResults) {
       if (getRequestId('searchWithdrawals') === requestId) {
-        setSearchResults(apiSearchResults);
+        setSearchResults(
+          apiSearchResults.map((result: any) => ({
+            ...result,
+            totalWithdrawn: numericParser.parse(result?.totalWithdrawn?.toString()),
```
[the remark this answers is not in the document]
Comment (karthikbtf) [reply src/components/NurseryWithdrawals/NurseryWithdrawals.tsx:213]: Yeah good point it should exist.

**Review by constanzauanini — APPROVED** [review]
Looks good in Vercel!

**Pull request merged**